client/src/screens/groups/components/GroupDetailsList.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Text,
+  FlatList,
+  Alert,
+} from 'react-native';
+
+import { Center, Container } from '../../../components/Container';
+
+import EventListItem from '../../../components/Events/EventListItem';
+import ScheduleListItem from '../../../components/Schedules/ScheduleListItem';
+import UserListItem from '../../../components/Users/UserListItem';
+
+
+const GroupDetailsRenderItem = ({ item }) => {
+  // eslint-disable-next-line no-underscore-dangle
+  switch (item.__typename) { // GQL typename from Schema
+    case 'User':
+      return (
+        <UserListItem
+          user={item}
+      // TODO: Deep link
+          onPress={() => Alert.alert('PLACEHOLDER', 'Deeplink to user details')}
+        />
+      );
+    case 'Schedule':
+      return (
+        <ScheduleListItem
+          schedule={item}
+    // TODO: Deep link
+          onPress={() => Alert.alert('PLACEHOLDER', 'Deeplink to schedule details')}
+        />
+      );
+    default:
+      return (
+        <EventListItem
+          event={item}
+    // TODO: Deep link
+          onPress={() => Alert.alert('PLACEHOLDER', 'Deeplink to event details')}
+        />
+      );
+  }
+};
+
+GroupDetailsRenderItem.propTypes = {
+  // eslint-disable-next-line react/forbid-prop-types
+  item: PropTypes.object.isRequired,
+};
+
+const GroupDetailsList = ({ items, selectedIndex, networkStatus, refetch }) => (
+  <Container>
+    <FlatList
+      data={items[selectedIndex]}
+      ListHeaderComponent={() =>
+        (!items.length ? (
+          <Center>
+            <Text>Nothing here</Text>
+          </Center>
+        ) : null)}
+      keyExtractor={item => item.id}
+      renderItem={item => GroupDetailsRenderItem(item)}
```

Comment:
a function that calls another function with the same parameter and returns the value from the function it calls is basically a no-op:
```
renderItem={groupDetailsRenderItem}
```
^ this should work exactly the same and saves the creation of extra anonymous functions

Reply:
ty

Reply:
im going to not action the button idea in this PR as there are a few places I can swap into a shared component that I will do a single PR for all of them